stan/math/prim/prob/wiener4_lcdf.hpp

```diff
@@ -0,0 +1,743 @@
+ * @return The log of the Wiener first passage time distribution with
+ *  the specified arguments for upper boundary responses
+ */
+template <bool propto = false, typename T_y, typename T_a, typename T_t0,
+          typename T_w, typename T_v>
+inline auto wiener_lcdf(const T_y& y, const T_a& a, const T_t0& t0,
+                        const T_w& w, const T_v& v,
+                        const double& precision_derivatives = 1e-4) {
+  using T_partials_return = partials_return_t<T_y, T_a, T_t0, T_w, T_v>;
+
+  using ret_t = return_type_t<T_y, T_a, T_t0, T_w, T_v>;
+  if (!include_summand<propto, T_y, T_a, T_t0, T_w, T_v>::value) {
+    return ret_t(0.0);
+  }
+
+  using T_y_ref = ref_type_if_t<!is_constant<T_y>::value, T_y>;
+  using T_a_ref = ref_type_if_t<!is_constant<T_a>::value, T_a>;
+  using T_t0_ref = ref_type_if_t<!is_constant<T_t0>::value, T_t0>;
+  using T_w_ref = ref_type_if_t<!is_constant<T_w>::value, T_w>;
+  using T_v_ref = ref_type_if_t<!is_constant<T_v>::value, T_v>;
+
+  static constexpr const char* function_name = "wiener4_lcdf";
+  if (size_zero(y, a, t0, w, v)) {
+    return ret_t(0.0);
+  }
+
+  check_consistent_sizes(function_name, "Random variable", y,
+                         "Boundary separation", a, "Drift rate", v,
+                         "A-priori bias", w, "Nondecision time", t0);
+
+  T_y_ref y_ref = y;
+  T_a_ref a_ref = a;
+  T_t0_ref t0_ref = t0;
+  T_w_ref w_ref = w;
+  T_v_ref v_ref = v;
+
+  decltype(auto) y_val = to_ref(as_value_column_array_or_scalar(y_ref));
+  decltype(auto) a_val = to_ref(as_value_column_array_or_scalar(a_ref));
+  decltype(auto) v_val = to_ref(as_value_column_array_or_scalar(v_ref));
+  decltype(auto) w_val = to_ref(as_value_column_array_or_scalar(w_ref));
+  decltype(auto) t0_val = to_ref(as_value_column_array_or_scalar(t0_ref));
+  check_positive_finite(function_name, "Random variable", y_val);
+  check_positive_finite(function_name, "Boundary separation", a_val);
+  check_finite(function_name, "Drift rate", v_val);
+  check_less(function_name, "A-priori bias", w_val, 1);
+  check_greater(function_name, "A-priori bias", w_val, 0);
+  check_nonnegative(function_name, "Nondecision time", t0_val);
+  check_finite(function_name, "Nondecision time", t0_val);
+
+  const size_t N = max_size(y, a, t0, w, v);
+
+  scalar_seq_view<T_y_ref> y_vec(y_ref);
+  scalar_seq_view<T_a_ref> a_vec(a_ref);
+  scalar_seq_view<T_t0_ref> t0_vec(t0_ref);
+  scalar_seq_view<T_w_ref> w_vec(w_ref);
+  scalar_seq_view<T_v_ref> v_vec(v_ref);
+  const size_t N_y_t0 = max_size(y, t0);
+
+  for (size_t i = 0; i < N_y_t0; ++i) {
+    if (y_vec[i] <= t0_vec[i]) {
+      std::stringstream msg;
+      msg << ", but must be greater than nondecision time = " << t0_vec[i];
+      std::string msg_str(msg.str());
+      throw_domain_error(function_name, "Random variable", y_vec[i], " = ",
+                         msg_str.c_str());
+    }
+  }
+
+  const auto log_error_cdf = log(1e-6);
```

Comment:
Is this a precision value users should be able to set?

Reply:
No, the user should not be able to set this value.